packages/repository/src/decorators/relation.decorator.ts

```diff
@@ -53,10 +54,22 @@ export function relation(definition?: Object) {
  * @param definition
  * @returns {(target:any, key:string)}
  */
-export function belongsTo(definition?: Object) {
-  // Apply model definition to the model class
-  const rel = Object.assign({type: RelationType.belongsTo}, definition);
-  return PropertyDecoratorFactory.createDecorator(RELATIONS_KEY, rel);
+export function belongsTo<T extends typeof Entity>(
+  targetModel: T | ModelResolver<T>,
```

Comment:
I want to change the relation decorator signature to something like this:
```ts
export function relation(
  targetOrDefinition: T | ModelResolver<T> | relationDefinitionBase,
  propertyDef: propertyDefinition
)
```
Thoughts?

Reply:
Oh oops, meant the propertyDef to be optional. I've fixed the OP with the edit